### ISSUE TITLE
Migrating from faulty database, probably a (too) old version: 

### ISSUE THREAD

**Issue by hillbicks** [RuudBurger/CouchPotatoServer#4598]
Hey guys,

a couple of weeks ago I realized that my  complete wanted list was gone. After a bit of research I decided to start from scratch, downloaded the zip from the github page, moved the ~/.couchpotato folder to a backup location and setup cp again, so there aren't any leftovers from previous installations.

This worked well until yesterday evening, when the wanted list was empty again. In addition to that, the profiles were reset (which happened before on a regular basis but didn't bother me much) The logfile shows the following:

01-28 23:08:22 INFO ESC[0m[tato.core.plugins.quality] Creating profile: BR-DiskESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.quality] Creating profile: 1080pESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.quality] Creating profile: 720pESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.quality] Creating profile: BR-RipESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.quality] Creating profile: DVD-RESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.quality] Creating profile: DVD-RipESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.quality] Creating profile: ScreenerESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.quality] Creating profile: R5ESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.quality] Creating profile: TeleCineESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.quality] Creating profile: TeleSyncESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.quality] Creating profile: CamESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.profile] Creating default profile: BestESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.profile] Creating default profile: HDESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.profile] Creating default profile: SDESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.profile] Creating default profile: Prefer 3D HDESC[0m
01-28 23:08:22 INFO ESC[0m[tato.core.plugins.profile] Creating default profile: 3D HDESC[0m
01-28 23:08:22 INFO ESC[0m[couchpotato.core.database] ==============================ESC[0m
01-28 23:08:22 INFO ESC[0m[couchpotato.core.database] Migrating database, hold on..ESC[0m
01-28 23:08:23 ERROR ESC[31m[couchpotato.core.database] Migrating from faulty database, probably a (too) old version: Traceback (most recent call last):
  File "/opt/CouchPotatoServer/couchpotato/core/database.py", line 326, in migrate
    c.execute('SELECT %s FROM `%s`' % ('`' + '`,`'.join(rows) + '`', ml))
OperationalError: no such table: profile
ESC[0m
01-28 23:08:23 INFO ESC[0m[couchpotato.core.database] Renaming old database to /home/hillbicks/.couchpotato/couchpotato.db.B3SMYACO_old ESC[0m
01-28 23:08:23 INFO ESC[0m[       couchpotato.runner] Starting server on port 5050ESC[0m

I think this is related to my deletion of some of the profile entries. (See the OperationalError: no such table: profile), but as of now, I couldn't reproduce the error.

Additionally, the file that is supposed to be the backup is empty and has a timestamp of 4 days before the logfile.

ll /home/hillbicks/.couchpotato/couchpotato.db.B3SMYACO_old
-rw-r--r-- 1 hillbicks hillbicks 0 Jan 24 04:50 /home/hillbicks/.couchpotato/couchpotato.db.B3SMYACO_old

Again, I'm fairly certain that this is related to the profiles. (cp sees they are incomplete and thinks something went wrong and it all goes downhill from there). This is not a 

I'll try and reproduce this, but this seems to be a bug in my opinion.

EDIT: Version 5b1dd686

**Comment by CouchPotatoBot** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Fill in all the dots. They're there for a reason
- [ ] Post logs, either inline (for smaller logs) or using [Pastebin](http://pastebin.com/)
- [ ] Post full version information

Please make sure you also read [contribution guide](https://github.com/RuudBurger/CouchPotatoServer/blob/develop/contributing.md#issues) and followed all the steps. 
Make the title describe your issue. Having "CP not working" or "I get this bug" for 100 issues, isn't really helpful. My master will close issues if there isn't enough information. On a good day he will tag the issue on close with the reason (like `can't reproduce`), but usually he won't, the lazy asshat.

Sometimes my master seems like a grumpy cat and responds with short answers. This isn't (always) because he hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let him know. Maybe he can teach me his awesome ways.

Thanks!

**Comment by RuudBurger** [comment]
It shouldn't migrate anymore after 1 time. Please remove the .db files from the data folder and restart.

**Comment by hillbicks** [comment]
Why should there even be a migration with  a _new_ version? And even if there was a need to upgrade the database, the backup didn't work. The backup file mentioned in the logfile is 
a) empty
b) has a timestamp from before the upgrade. 

Well, I just had this issue again.

Restarted CouchpotatoServer and now the wanted list is empty, with the same error in the logfile:

01-31 11:29:18 INFO ESC[0m[couchpotato.core.database] ==============================ESC[0m
01-31 11:29:18 INFO ESC[0m[couchpotato.core.database] Migrating database, hold on..ESC[0m
01-31 11:29:19 ERROR ESC[31m[couchpotato.core.database] Migrating from faulty database, probably a (too) old version: Traceback (most recent call last):
  File "/opt/CouchPotatoServer/couchpotato/core/database.py", line 326, in migrate
    c.execute('SELECT %s FROM `%s`' % ('`' + '`,`'.join(rows) + '`', ml))
OperationalError: no such table: profile
ESC[0m
01-31 11:29:19 INFO ESC[0m[couchpotato.core.database] Renaming old database to /home/hillbicks/.couchpotato/couchpotato.db.XOMUCLKS_old ESC[0m


And agan, the backup file is empty and has a timestamp from two days ago. As I said earlier, I removed the db files before I started from scratch with the version from 23rd of January. I'm more than happy to provide more info and or configs that I'm using, just let me know.

**Comment by hillbicks** [comment]
I just checked again and saw that I don't even have a couchpotato.db file anymore in addition to the old one being empty, even after several restarts. I guess this is the .db file you're taling about?

There is a database folder, but that's it.

**Comment by RuudBurger** [comment]
Are you in the correct folder? It only does a migrate when it sees an db file named couchpotato.db

**Comment by hillbicks** [comment]
yep, definitely. I did a locate couchpotato.db which showed results from the backup I made and in the current directory ~/.couchpotato/couchpotato.db but the file wasn't there anymore. locate only showed it because the last update was made before the file was deleted. 

I also created a new folder in my homedir (couch), set it as the datadir in the settings and restart couch. 4 directories were created (cache, custom_plugins, database, logs). No other files in that dir. 

These are the ERRORS from the logfile in the new dir

01-31 12:05:34 ERROR [tato.core.plugins.renamer] Missing `cd` or `cd_nr` in the renamer. This will cause multi-file releases of being renamed to the same file. Please add it in the renamer settings. Force adding it for now.
01-31 12:05:47 ERROR [tato.core.plugins.scanner] No imdb_id found for [u' dvdr', u'']. Add a NFO file with IMDB id or add the year to the filename.
01-31 12:05:47 ERROR [tato.core.plugins.scanner] Unable to determine media: [u' dvdr', u'']
01-31 12:05:55 ERROR [hpotato.core.plugins.base] Failed opening url in OMDBAPI: http://www.omdbapi.com/?i=tt1309447 Traceback (most recent call last):

**Comment by hillbicks** [comment]
oh man..... :P

I set up a cronjob to check for the existence of the couchpotato.db file and I just got a mail that the file exists. After digging around some, I figured out that I set up a script when transmission finishes a file, a script is called. In this script I would try and extract the imdb id from couchpotato.db with sqlite3, this caused the file to be created empty which in turn triggered couchpotato to try and migrate the file, which failed and was ultimately the cause for the emtpy wanted list. 

Sorry about that, classic pebkac :)
